lib/ecto/adapters/postgres/connection.ex

```diff
@@ -253,10 +253,18 @@ if Code.ensure_loaded?(Postgrex) do
     @impl true
     def explain_query(conn, query, params, opts) do
       {explain_opts, opts} =
-        Keyword.split(opts, ~w[analyze verbose costs settings buffers timing summary]a)
+        Keyword.split(opts, ~w[analyze verbose costs settings buffers timing summary format]a)
+
+      json_format? = {:format, :json} in explain_opts
 
       case query(conn, build_explain_query(query, explain_opts), params, opts) do
-        {:ok, %Postgrex.Result{rows: rows}} -> {:ok, Enum.map_join(rows, "\n", & &1)}
+        {:ok, %Postgrex.Result{rows: rows}} when json_format? ->
+          case Jason.encode(List.flatten(rows)) do
+            {:ok, json} -> {:ok, json}
+            {:error, error} -> error
+          end
```

Comment:
I thought we agreed on returning text for all of them? :)

Reply:
@josevalim not sure if I understood you. You meant to return error as text or the success result? The tuple value is eventually extracted later on. I'm not sure about returning the custom text for error, maybe we can just it as it is?

Reply:
Oh, please ignore me. You are calling *Jason.encode*.

Hrm, I wonder if there is a way to ask Postgrex to not decode the data in the first place? Jason is unfortunately an optional dependency, so we cannot assume it is available. Or we use `format: :map` (which behind the scenes translates `format: :json`).